media/main.js

```diff
@@ -85,5 +89,37 @@
         }
     }
 
-})();
+    async function saveSvg() {
+        const svgElement = document.querySelector("#mermaid-diagram svg");
+        if (!svgElement) {
+            console.error("No SVG element found");
+            return;
+        }
+
+        const serializer = new XMLSerializer();
+        const svgString = serializer.serializeToString(svgElement);
+        const blob = new Blob([svgString], { type: "image/svg+xml" });
+        const url = URL.createObjectURL(blob);
 
+        const a = document.createElement("a");
+        a.href = url;
+        a.download = await generateFileName();
+        document.body.appendChild(a);
+        a.click();
+        document.body.removeChild(a);
+        URL.revokeObjectURL(url);
+    }
+
+    async function generateFileName() {
+        const date = new Date();
+        const timestamp = date.toISOString().replace(/[:.-]/g, "");
+        const prompt = `Generate a reasonable file name for an SVG diagram. Current timestamp: ${timestamp}`;
+        const response = await vscode.commands.executeCommand('vscode.executeLanguageModel', {
```

Comment:
I'm not sure this will work.

Reply:
Should handle similarly to https://github.com/microsoft/vscode-mermAId/blob/23f60a993bbe10e5db47255af3ab000f534d5c1a/src/outlineView.ts#L454